src/Validation/Validation.php

```diff
@@ -1082,6 +1082,76 @@ public static function uploadedFile($file, array $options = [])
     }
 
     /**
+     * Validates the size of an uploaded image.
+     *
+     * @param array $file The uploaded file data from PHP.
+     * @param array $options Options to validate width and height.
+     * @return bool
+     */
+    public static function imageSize($file, $options)
+    {
+        if (!isset($options['height']) && !isset($options['width'])) {
+            throw new InvalidArgumentException('Invalid image size validation parameters! Missing `width` and / or `height`!');
+        }
+
+        list($width, $height) = getimagesize($file['tmp_name']);
```

Comment:
I think it would be good if this could handle non-uploads too, like for example [`mimeType()`](https://github.com/burzum/cakephp/blob/bba091dc55976b04c7537943aa0163a3b3a5d7f9/src/Validation/Validation.php#L955-L957).